src/syscheckd/fim_db.c

```diff
@@ -528,27 +531,34 @@ int fim_db_exec_simple_wquery(fdb_t *fim_sql, const char *query) {
 }
 
 int fim_db_sync_path_range(fdb_t * fim_sql, pthread_mutex_t *mutex, fim_tmp_file *file, int storage) {
-    return fim_db_process_read_file(fim_sql, file, mutex, fim_db_callback_sync_path_range,
-            storage, (void *) (int) 0);
+    return fim_db_process_read_file(fim_sql, file, mutex, fim_db_callback_sync_path_range, storage,
+                                    NULL, NULL, NULL);
 }
 
 int fim_db_delete_not_scanned(fdb_t * fim_sql, fim_tmp_file *file, pthread_mutex_t *mutex, int storage) {
-    return fim_db_process_read_file(fim_sql, file, mutex, fim_db_remove_path,
-            storage, (void *) (int) 1);
+    return fim_db_process_read_file(fim_sql, file, mutex, fim_db_remove_path, storage,
+                                    NULL, NULL, (void *) (int) true);
 }
 
 int fim_db_delete_range(fdb_t * fim_sql, fim_tmp_file *file, pthread_mutex_t *mutex, int storage) {
-    return fim_db_process_read_file(fim_sql, file, mutex, fim_db_remove_path,
-            storage, (void *) (int) 0);
+    return fim_db_process_read_file(fim_sql, file, mutex, fim_db_remove_path, storage,
+                                    NULL, NULL, (void *) (int) false);
 }
 
-int fim_db_process_missing_entry(fdb_t *fim_sql, fim_tmp_file *file, pthread_mutex_t *mutex, int storage, fim_event_mode mode) {
-    return fim_db_process_read_file(fim_sql, file, mutex, fim_db_process_path, storage, (void *) (fim_event_mode) mode);
+int fim_db_process_missing_entry(fdb_t *fim_sql, fim_tmp_file *file, pthread_mutex_t *mutex, int storage,
+                                    fim_event_mode mode, whodata_evt * w_evt) {
+    return fim_db_process_read_file(fim_sql, file, mutex, fim_db_process_path, storage, (void *) (fim_event_mode) mode,
+                                    (void *) w_evt, NULL);
 }
 
+/* 
+    TODO:
+        - Move mode, w_evt and send_alert to an array to not send so many unused variables to
+        the functions that don't need them.
+ */
 int fim_db_process_read_file(fdb_t *fim_sql, fim_tmp_file *file, pthread_mutex_t *mutex,
```

Comment:
For simplicity's sake, we should add all these optional parameters(`send_alert`, `wet` and `mode`) into an array. With this, the code would be much more readable and modular.